src/pythermalcomfort/zhang_comfort_model_functions/sensation.py

```diff
@@ -0,0 +1,244 @@
+        # Clip the array between -4 and 4
+        sensation_local_array = np.clip(sensation_local_array, -4, 4)
+    else:
+        raise ValueError("Please select '7-point' or '9-point'")
+
+    return sensation_local_array
+
+
+def calculate_overall_sensation(sensation_local_array):
+    """Calculates overall sensation.
+
+    Parameters:
+    sensation_local_array (array): Array of sensations from various body parts.
+
+    :returns
+    overall_sensation_dict (dict): Dict ....
+
+    """
+
+    # Define local thermal comfort as a dictionary
+    local_17_segments_dict = utilities.convert_17_segments_dict_from_array(
+        array=sensation_local_array, body_name_list=utilities.body_name_list
+    )
+
+    def are_most_extreme_sensations_hands_or_feet(
+            sensation_local_array,
+            left_hand_sensation,
+            right_hand_sensation,
+            left_foot_sensation,
+            right_foot_sensation,
+    ):
+        """Determine if the most extreme sensations in a given array are from the hands or feet.
+
+        This function examines the input array of sensations to determine if the most
+        extreme sensations (both maximum and minimum) come from the hands or feet. The
+        sensations of the hands and feet are provided separately for comparison.
+
+        Parameters
+        ----------
+        sensation_local_array : list or ndarray
+            An array or list containing sensations from various body parts.
+        left_hand_sensation : float
+            The sensation value corresponding to the left hand.
+        right_hand_sensation : float
+            The sensation value corresponding to the right hand.
+        left_foot_sensation : float
+            The sensation value corresponding to the left foot.
+        right_foot_sensation : float
+            The sensation value corresponding to the right foot.
+
+        Returns
+        -------
+        bool
+            Returns True if the most extreme sensations (either the two largest or two smallest)
+            correspond to both hands or both feet. Otherwise, returns False.
+
+        Notes
+        -----
+        The function assumes that the `sensation_local_array` contains at least two elements.
+        If not, it will directly return False without further checks.
+
+        """
+
+        # Sensation array must have at least two elements for this function to work properly
+        if len(sensation_local_array) < 2:
+            return False
+
+        # Extract the top two maximum and minimum sensations
+        max_two = sorted(sensation_local_array, reverse=True)[:2]
+        min_two = sorted(sensation_local_array)[:2]
+
+        # Set up hand and foot sensations
+        hand_sensations = {left_hand_sensation, right_hand_sensation}
+        foot_sensations = {left_foot_sensation, right_foot_sensation}
+
+        # Check if the most extreme sensations correspond to two hands or two feet
+        return (
```

Comment:
could you please reformat all the files with Black, the formatting here is very strange

Reply:
Reformatted.